dynomite-derive/src/lib.rs

```diff
@@ -146,14 +146,18 @@ fn make_dynomite_attr(
 }
 
 fn expand_item(ast: DeriveInput) -> syn::Result<impl ToTokens> {
+    use syn::spanned::Spanned as _;
     let name = &ast.ident;
     let vis = &ast.vis;
     match ast.data {
         Struct(DataStruct { fields, .. }) => match fields {
             Fields::Named(named) => {
                 make_dynomite_item(vis, name, &named.named.into_iter().collect::<Vec<_>>())
             }
-            _ => panic!("Dynomite Items require named fields"),
+            fields => Err(syn::Error::new(
```

Comment:
also took some liberty to improve error ux. this change gives the user a precise location of the source of the problem